Makefile

```diff
@@ -155,7 +155,7 @@ test:
 
 test-smoke:
 	$(info log-file: test-$(TEST_SUFFIX).log)
-	PASSES="fmt build unit" ./test.sh 2<&1 | tee test-$(TEST_SUFFIX).log
+	PASSES="fmt bom dep build unit" ./test.sh 2<&1 | tee test-$(TEST_SUFFIX).log
```

Comment:
We no longer use `test-smoke` on main branch.
Please adopt naming and conventions from main branch.

Reply:
@serathius the goal of this PR is to have the missing `fmt` job added back to the pipeline, there should be another over arching issue trying to closely mimic the `main` pipeline as discussed in Feb 15th 2024 Bug Triage meeting.

Please correct me if I'm wrong. @jmhbnz @ahrtr @ivanvc

Reply:
Link to discussion: https://youtu.be/p7K8EH3jayo?t=1303.  This pr is the minimal pr to get basic lint checks running in `release-3.5`.  Follow-up pull requests will backport our structure of `make verify` to `release-3.5` to create parity between the two branches.

Reply:
The only minor concern is the naming `test-smoke `is a little confusing. But it isn't introduced in this PR, so it's OK to keep it as it's or resolve it in a separate PR.